tests/lora/test_lora_model.py

```diff
@@ -0,0 +1,198 @@
+    qeff_model = QEffAutoLoraModelForCausalLM(model_hf, pretrained_model_name_or_path=base_model_name)
+
+    assert qeff_model.base_model_name == base_model_name
+    assert len(qeff_model.adapter_weights) == 0
+    assert len(qeff_model.adapter_configs) == 0
+    assert len(qeff_model.active_adapters) == 0
+    assert qeff_model.max_num_adapters == 0
+    assert len(qeff_model.active_adapter_to_id) == 0
+
+
+@pytest.mark.parametrize("base_model_name,adapter_id_0,adapter_id_1", model_samples)
+def test_auto_lora_model_for_causal_lm_from_pretrained(base_model_name, adapter_id_0, adapter_id_1):
+    qeff_model = QEffAutoLoraModelForCausalLM.from_pretrained(base_model_name, num_hidden_layers=1)
+
+    assert qeff_model.base_model_name == base_model_name
+    assert len(qeff_model.adapter_weights) == 0
+    assert len(qeff_model.adapter_configs) == 0
+    assert len(qeff_model.active_adapters) == 0
+    assert qeff_model.max_num_adapters == 0
+    assert len(qeff_model.active_adapter_to_id) == 0
+
+
+def test_auto_lora_model_for_causal_lm_hash():
+    base_config_0, adapter_config_0 = configs[0].values
+    base_config_1, adapter_config_1 = configs[1].values
+
+    qeff_model_0 = create_lora_base_model(base_config_0)
+    qeff_model_0 = load_adapter_with_random_weights(qeff_model_0, adapter_config_0, "adapter_0")
+    qeff_model_0 = load_adapter_with_random_weights(qeff_model_0, adapter_config_1, "adapter_1")
+
+    qeff_model_1 = create_lora_base_model(base_config_1)
+    qeff_model_1 = load_adapter_with_random_weights(qeff_model_1, adapter_config_0, "adapter_0")
+    qeff_model_1 = load_adapter_with_random_weights(qeff_model_1, adapter_config_1, "adapter_1")
+
+    # check num of adapter config matters
+    qeff_model_0.set_adapter("adapter_0")
+    model_hash_0_0 = qeff_model_0.model_hash
+    qeff_model_0.set_adapter("adapter_1")
+    model_hash_0_1 = qeff_model_0.model_hash
+    assert model_hash_0_0 != model_hash_0_1
+
+    # check base model configs matters
+    qeff_model_1.set_adapter("adapter_0")
+    qeff_model_1.set_adapter("adapter_1")
+    model_hash_1_1 = qeff_model_1.model_hash
+    assert model_hash_1_1 != model_hash_0_1
+
+    # check adapter orders matters
+    qeff_model_1.delete_adapter("adapter_0")
+    qeff_model_1.delete_adapter("adapter_1")
+    qeff_model_1.set_adapter("adapter_1")
+    qeff_model_1.set_adapter("adapter_0")
+    model_hash_1_2 = qeff_model_1.model_hash
+    assert model_hash_1_2 != model_hash_1_1
+
+    # check if different adapter config, but same adapter name matters
+    qeff_model_0 = load_adapter_with_random_weights(qeff_model_0, adapter_config_0, "adapter_1")
+    model_hash_0_2 = qeff_model_0.model_hash
+    assert model_hash_0_2 != model_hash_0_1
+
+
+@pytest.mark.parametrize("base_model_name,adapter_id_0,adapter_id_1", model_samples[:1])
+def test_auto_lora_model_for_causal_lm_load_set_adapter(base_model_name, adapter_id_0, adapter_id_1):
```

Comment:
add some description and add id_check in the name.

Reply:
added